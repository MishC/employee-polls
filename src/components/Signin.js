// Signin.js
import React, { useRef, useState } from 'react';
import { connect } from 'react-redux';
import { setAuthedUser } from '../actions/authedUser';

function Signin({ userSetUp }) {
  const [error, setError] = useState(null);

  const id = useRef('');
  const password = useRef('');

  const handleSubmit = (e) => {
    e.preventDefault(); 
    if (id.current && password.current) {
      userSetUp(id.current.value, password.current.value);
    }
    else setError("Something went wrong. Try Again")
  };

  return (
    <div>
      <form onSubmit={handleSubmit}>
        <input
          type="text"
          ref={id}
          placeholder="Enter your ID"
        />
        <input
          type="password"
          ref={password}
          placeholder="Enter your password"
        />
        <button type="submit">Submit</button>
      </form>
      {error && <div style={{ color: 'red' }}>{error}</div>}
    </div>
  );
}

export default Signin;
