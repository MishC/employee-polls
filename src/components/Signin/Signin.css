.Signin {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    height: 100vh;
    background-color:var(--main-bg-color);
  }
  
  .signin-input-container {
    display: flex;
    margin-bottom: 20px;
  }
  
  .signin-input {
    flex: 1; 
    height: 40px;
    margin: 10px;
    padding: 8px;
    font-size: 16px;
    border: 1px solid var(--shadow-grey);
    border-radius: 4px;
    transition: border-color 0.3s ease-in-out;
  }
  
  .signin-input:focus {
    border-color: dodgerblue; }
  
  .signin-button {
    width: 120px; 
    height: 50px; 
    margin: 10px;
    padding: 8px;
    font-size: 16px;
    background-color:var(--main-color); 
    color: white; 
    border: none;
    border-radius: 4px;
    cursor: pointer;
  }
  
  .signin-button:hover {
    opacity: 0.8; 
    cursor: pointer;

  }
  
  .error-message {
    color: red;
    margin-top: 10px;
  }
  