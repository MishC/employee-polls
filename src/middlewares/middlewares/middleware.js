import {thunk} from "redux-thunk";
import logger from "./logger";
import authCheck from "./authCheck";

const middleware = [thunk, logger];

export default middleware;
