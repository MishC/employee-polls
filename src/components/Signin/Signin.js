import React, { useRef, useState } from 'react';
import { connect } from 'react-redux';
import './Signin.css';

function Signin({ userSetUp }) {
  const [error, setError] = useState(null);

  const id = useRef('');
  const password = useRef('');

  const handleSubmit = (e) => {
    e.preventDefault(); 
    if (id.current && password.current) {
      userSetUp(id.current.value, password.current.value);
    }
  };

  return (
    <div className="Signin">
      <form onSubmit={handleSubmit}  className="Signin">
        <div className="signin-input-container">
          <input
            type="text"
            ref={id}
            className="signin-input"
            placeholder="Enter your ID"
          />
          <input
            type="password"
            ref={password}
            className="signin-input"
            placeholder="Enter your password"
          />
        </div>
        <button type="submit" className="signin-button">Submit</button>
      </form>
      {error && <div className="error-message">{error}</div>}
    </div>
  );
}

export default connect()(Signin);
