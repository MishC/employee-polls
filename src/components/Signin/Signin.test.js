// src/components/MyComponent.test.js
import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { configureStore } from '@reduxjs/toolkit';
import reducer from "../../reducers";

import Signin from './Signin'; 
import middleware from '../../middlewares/middleware'; 

// Create a mock store

const mock_store = configureStore({
  reducer,
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(middleware),
});


describe('Signin', () => {
  it('renders correctly', () => {
    const { asFragment } = render(
      <Provider store={mock_store}>
        <Signin />
      </Provider>
    );

    // Snapshot testing
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders input password', () => {
    const { getByTestId } = render(
      <Provider store={mock_store}>
        <Signin />
      </Provider>
    );

    expect(getByTestId('psw')).toBeInTheDocument();
  });
  it('renders input id', () => {
    const { getByTestId } = render(
      <Provider store={mock_store}>
        <Signin />
      </Provider>
    );

    expect(getByTestId('id')).toBeInTheDocument();
  });

  it('renders submit button', () => {
    const { getByText } = render(
      <Provider store={mock_store}>
        <Signin />
      </Provider>
    );

    expect(getByText(/Submit/i)).toBeInTheDocument();
  });

  // Add more tests to verify different aspects of your component
});
