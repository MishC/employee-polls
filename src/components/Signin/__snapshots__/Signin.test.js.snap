// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signin renders correctly 1`] = `
<DocumentFragment>
  <div
    class="Signin"
  >
    <form
      class="Signin"
    >
      <div
        class="signin-input-container"
      >
        <input
          class="signin-input"
          data-testid="id"
          placeholder="Enter your ID"
          type="text"
        />
        <input
          class="signin-input"
          data-testid="psw"
          placeholder="Enter your password"
          type="password"
        />
      </div>
      <button
        class="signin-button"
        data-testid="submit"
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
